lib/assignment-visitor.js

```diff
@@ -1,84 +1,79 @@
 "use strict";
 
 const utils = require("./utils.js");
-const hasOwn = Object.prototype.hasOwnProperty;
 const Visitor = require("./visitor.js");
 
-function AssignmentVisitor() {
-  Visitor.call(this);
-}
-
-module.exports = AssignmentVisitor;
-
-const AVp = AssignmentVisitor.prototype =
-  Object.create(Visitor.prototype);
-AVp.constructor = AssignmentVisitor;
+module.exports = class AssignmentVisitor extends Visitor {
+  constructor() {
+    super()
+  }
```

Comment:
This constructor shouldn't be necessary, since `Visitor.prototype.constructor` will get called directly without it.